openedx/features/content_tagging/rest_api/v1/tests/test_views.py

```diff
@@ -0,0 +1,964 @@
+        ("userA", "tB1", status.HTTP_200_OK),
+        ("userA", "tB2", status.HTTP_404_NOT_FOUND),
+        ("userA", "tC1", status.HTTP_200_OK),
+        ("userA", "tC2", status.HTTP_200_OK),
+        ("userC", "tA1", status.HTTP_200_OK),
+        ("userC", "tA2", status.HTTP_200_OK),
+        ("userC", "tB1", status.HTTP_200_OK),
+        ("userC", "tB2", status.HTTP_200_OK),
+        ("userC", "tC1", status.HTTP_200_OK),
+        ("userC", "tC2", status.HTTP_200_OK),
+        ("userS", "st1", status.HTTP_200_OK),
+        ("userS", "st2", status.HTTP_200_OK),
+        ("userS", "t1", status.HTTP_200_OK),
+        ("userS", "t2", status.HTTP_200_OK),
+        ("userS", "tA1", status.HTTP_200_OK),
+        ("userS", "tA2", status.HTTP_200_OK),
+        ("userS", "tB1", status.HTTP_200_OK),
+        ("userS", "tB2", status.HTTP_200_OK),
+        ("userS", "tC1", status.HTTP_200_OK),
+        ("userS", "tC2", status.HTTP_200_OK),
+    )
+    @ddt.unpack
+    def test_detail_taxonomy(self, user_attr, taxonomy_attr, expected_status):
+        taxonomy = getattr(self, taxonomy_attr)
+
+        url = TAXONOMY_ORG_DETAIL_URL.format(pk=taxonomy.id)
+
+        if user_attr:
+            user = getattr(self, user_attr)
+            self.client.force_authenticate(user=user)
+
+        response = self.client.get(url)
+        assert response.status_code == expected_status
+
+        if status.is_success(expected_status):
+            check_taxonomy_object(response.data, taxonomy)
+
+    @ddt.data(
+        (None, status.HTTP_403_FORBIDDEN),
+        ("userA", status.HTTP_403_FORBIDDEN),
+        ("userB", status.HTTP_403_FORBIDDEN),
+        ("userC", status.HTTP_403_FORBIDDEN),
+        ("userS", status.HTTP_201_CREATED),
+    )
+    @ddt.unpack
+    def test_create_taxonomy(self, user_attr, expected_status):
+        url = TAXONOMY_ORG_LIST_URL
+
+        create_data = {
+            "name": "taxonomy_data",
+            "description": "This is a description",
+            "enabled": False,
+            "required": True,
+            "allow_multiple": True,
+        }
+
+        if user_attr:
+            user = getattr(self, user_attr)
+            self.client.force_authenticate(user=user)
+
+        response = self.client.post(url, create_data, format="json")
+        assert response.status_code == expected_status
+
+        # If we were able to create the taxonomy, check if it was created
+        if status.is_success(expected_status):
+            check_taxonomy(response.data, response.data["id"], **create_data)
+            url = TAXONOMY_ORG_DETAIL_URL.format(pk=response.data["id"])
+
+            response = self.client.get(url)
+            check_taxonomy(response.data, response.data["id"], **create_data)
+
+    @ddt.data(
+        (None, "t1", status.HTTP_403_FORBIDDEN),
+        ("user", "t1", status.HTTP_403_FORBIDDEN),
+        ("user", "tA1", status.HTTP_403_FORBIDDEN),
+        ("userA", "tA1", status.HTTP_200_OK),
```

Comment:
Can you add more checks for the instance-wide taxonomies too?  `userA` should be forbidden from updating/patching `t1` and `t2`.

Reply:
Done